xbmc/Application.cpp

```diff
@@ -1659,6 +1659,19 @@ bool CApplication::OnAction(const CAction &action)
     }
     return true;
   }
+  // Tone Mapping : switch to next tone map method
+  if (action.GetID() == ACTION_CYCLE_TONEMAP_METHOD)
+  {
+    if (m_appPlayer.IsPlayingVideo())
+    {
+      CVideoSettings vs = m_appPlayer.GetVideoSettings();
+      vs.m_ToneMapMethod++;
+      if (vs.m_ToneMapMethod >= VS_TONEMAPMETHOD_MAX)
+        vs.m_ToneMapMethod = VS_TONEMAPMETHOD_OFF + 1;
+      m_appPlayer.SetVideoSettings(vs);
+    }
+    return true;
+  }
```

Comment:
Mhm, should we handle this in https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L4107 instead?

Reply:
It's okay here I think. The player should basically not have to know about settings. So that application player sets everything up and then the real player can just continue to play?